feat(signals): enhance `withComputed` to accept computation functions (#4822)
ngrx/platform: projects/ngrx.io/content/guide/signals/signal-store/index.md

Computed signals can be added to the store using the `withComputed` feature.
This feature accepts a factory function as an input argument, which is executed within the injection context.

> The factory should return a dictionary containing either computed signals or functions that return values (which will automatically be wrapped in computed signals), utilizing previously defined state signals and properties that are accessible through its input argument.

Reviewer: ```suggestion
The factory should return a dictionary containing either computed signals or computation functions (which are automatically converted to computed signals), utilizing previously defined state signals and properties that are accessible through its input argument.
```

Author: @markostanimirovic @timdeschryver, do we want to use the term "computation functions"? I am afraid that this could confuse users.

> The factory should return a dictionary containing either computed signals or functions that return values (which will automatically be wrapped in computed signals), utilizing previously defined state signals and properties that are accessible through its input argument.

Author: Done
